lib/vector-core/core-common/src/internal_event/events_sent.rs

```diff
@@ -2,22 +2,35 @@ use metrics::counter;
 
 use crate::internal_event::InternalEvent;
 
+pub const DEFAULT_OUTPUT: &str = "_default";
+
 #[derive(Debug)]
-pub struct EventsSent {
+pub struct EventsSent<'a> {
     pub count: usize,
     pub byte_size: usize,
+    pub output: Option<&'a str>,
 }
 
-impl InternalEvent for EventsSent {
+impl<'a> InternalEvent for EventsSent<'a> {
     fn emit_logs(&self) {
-        trace!(message = "Events sent.", count = %self.count, byte_size = %self.byte_size);
+        if let Some(output) = self.output {
+            trace!(message = "Events sent.", count = %self.count, byte_size = %self.byte_size, output = %output);
+        } else {
+            trace!(message = "Events sent.", count = %self.count, byte_size = %self.byte_size);
+        }
     }
 
     fn emit_metrics(&self) {
         if self.count > 0 {
-            counter!("component_sent_events_total", self.count as u64);
-            counter!("events_out_total", self.count as u64);
-            counter!("component_sent_event_bytes_total", self.byte_size as u64);
+            if let Some(output) = self.output {
+                counter!("component_sent_events_total", self.count as u64, "output" => output.to_owned());
+                counter!("events_out_total", self.count as u64, "output" => output.to_owned());
+                counter!("component_sent_event_bytes_total", self.byte_size as u64, "output" => output.to_owned());
```

Comment:
Do we need the `to_owned` calls here? It'd be nice to avoid the cost of these extra allocations.

Reply:
I get a lifetimes error when trying to use the reference which I think boils down to `counter!` requiring `lifetime must be valid for the static lifetime`, so I opted for `to_owned` here. Is there a nicer alternative?

Reply:
Ah, maybe not. That's probably fine for now and we can do something clever if it shows up as meaningful when we're profiling.

Reply:
Confirmed, all the tags on the output metrics need to be valid for the `static` lifetime.